maintenance/checkSVGs.php

```diff
@@ -0,0 +1,155 @@
+
+		$this->db = $this->getDBConnection();
+		$this->upload = new UploadDummy;
+		$total = 0;
+		$bad = 0;
+		$errors = 0;
+
+		$repo = RepoGroup::singleton()->getLocalRepo();
+		$fileBackend = $repo->getBackend();
+
+		while ( $candidates = $this->getCandidates() ) {
+			foreach( $candidates as $candidate ) {
+				$total++;
+				$file = $repo->newFileFromRow( $candidate );
+
+				// debugging info
+				$globalTitle = GlobalTitle::newFromText( $file->getTitle()->getText(), $file->getTitle()->getNamespace(), $wgCityId );
+				$details = $globalTitle->getFullURL() . " (" . $file->getUrl() . ")";
+
+				$path = $file->getPath();
+				if (!$path) {
+					$this->output("\tError while getting path for ${details}\n");
+					$errors++;
+					continue;
+				}
+
+				$fileContents = $fileBackend->getFileContents( array( 'src' => $path ) );
+				if (!$fileContents) {
+					$this->output("\tError while getting contents of ${details}\n");
+					$errors++;
+					continue;
+				}
+
+				$tmpFile = tempnam( wfTempDir(), 'img' );
+				file_put_contents($tmpFile, $fileContents);
+				$res = $this->checkFile( $tmpFile );
+				unlink($tmpFile);
+				if ( $res !== true ) {
+					$bad++;
+					$this->output("\tValidation error '${res}' while checking ${details}\n");
+				}
+			}
+			$this->output("Done batch ${total} - at: " . $candidates[count($candidates)-1]->img_name . "\n");
+		}
+		$this->output("Complete: ${total} total; ${bad} bad, ${errors} errors\n");
+	}
+
+	private function checkFile( $file ) {
+		return $this->upload->checkFile( $file );
+	}
+
+	private function getCandidates() {
+		$conds = [
+			'img_size < ' . ( (int)$this->getOption( 'max-size', 1024*1024*10 ) ),
+			'img_major_mime' => 'image',
+			'img_minor_mime' => 'svg+xml',
+			'img_media_type' => 'DRAWING'
+		];
+		if ( $this->lastFileName ) {
+			$conds[] = 'img_name > ' . $this->db->addQuotes( $this->lastFileName );
+		}
+
+		$until = $this->getOption( 'until' );
+		if ( $until ) {
+			$conds[] =
+				'img_timestamp < ' . $this->db->addQuotes( $until );
+		}
+
+		$res = $this->db->select(
+			'image',
+			'*',
+			$conds,
+			__METHOD__,
+			[
+				'ORDER BY' => 'img_name asc',
+				'LIMIT' => $this->mBatchSize
+
+			]
+		);
+
+		$actualResults = [];
+		foreach ( $res as $row ) {
+			$actualResults[] = $row;
+			$this->lastFileName = $row->img_name;
+		}
+		return $actualResults;
+	}
+}
+
+class UploadDummy extends UploadBase {
+	public function initializeFromRequest( &$request ) {}
+
+	/**
+	 * @param String Full contents of svg file
+	 *
+	 * @return true if ok, or string for error code
+	 */
+	public function checkFile( $file ) {
+		$this->mSVGNSError = false;
+		$check = new XmlTypeCheck(
+			$file,
+			[ $this, 'checkSvgScriptCallback' ],
+			[ 'processing_instruction_handler' => 'UploadBase::checkSvgPICallback' ]
+		);
+		if ( $check->wellFormed !== true ) {
+			return 'uploadinvalidxml';
+		} elseif ( $check->filterMatch ) {	// validation error
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)